src/zenml/utils/source_utils.py

```diff
@@ -69,6 +62,8 @@
 )
 
 _SHARED_TEMPDIR: Optional[str] = None
+_resolved_notebook_sources: Dict[str, str] = {}
+_notebook_modules: Dict[str, UUID] = {}
```

Comment:
Just a small nitpick here, I think the name `notebook_modules` is slightly misleading as it maps modules to artifact store IDs. But, since it's an internal variable, feel free to ignore/resolve this as you wish.